internal/config/sshSetup.go

```diff
@@ -107,6 +107,9 @@ func SetAuth(configuration *Config) error {
 
 		default:
 			// Set custom key if specified
+                        if strings.HasPrefix(node.Auth.Keys.Path, "~") {
+                           strings.Replace(node.Auth.Keys.Path, "~", homePath, 1)
```

Comment:
the `strings.Replace` returns the modified string instead of changing the string in-place. So you will have to assign this back to `node.Auth.Keys.Path`.
```suggestion
                           node.Auth.Keys.Path = strings.Replace(node.Auth.Keys.Path, "~", homePath, 1)
```